Added missing functions to create Boundaries

Feuerbowle opened the pull request:
Added functions to create:
- Block Boundary
- Selected Surface Boundary
- Shallow Boundary
- Collision Safe Boundary
- Contact Point Boundary from Boundary, Pattern, Toolpath and Model
- Contact Conversion Boundary
- Boolean Operation Boundary
    for this boundary function, i added a macro for testing that creates two overlapping boundaries and also a new enum for the boolean types.

I will add the function to create Stock Model Rest Boundary later, because I first have to extend the Stockmodel class to have access to the Stockmodel states.

Feuerbowle committed:
Added missing functions to create Boundaries

Added functions to create:
- Block Boundary
- Selected Surface Boundary
- Shallow Boundary
- Collision Safe Boundary
- Contact Point Boundary from Boundary, Pattern, Toolpath and Model
- Contact Conversion Boundary
- Boolean Operation Boundary
    for this boundary function, i added a macro for testing that creates two overlapping boundaries and also a new enum for the boolean types.

I will add the function to create Stock Model Rest Boundary later, because I first have to extend the Stockmodel class to have access to the Stockmodel states.

Feuerbowle committed:
Extend the StockModel Class

Added functions to:
- apply the block, toolpath (first/last) and tool (first/last) to the stockmodel
- get the names of all stockmodel states as string list

lukeedw commented:
@Feuerbowle looks great so far.  Let me know when you are ready for me to pull it (I noticed you just made another commit).

Feuerbowle commented:
> 
> 
> @Feuerbowle looks great so far. Let me know when you are ready for me to pull it (I noticed you just made another commit).

Ok, I will let you know when I'm ready. I think I will finish the missing StockModel Rest Boundary function by tomorrow if nothing comes between.

Feuerbowle committed:
Added function for StockModel Rest Boundary

here is finally the last missing Boundary creation function.
Now the "boundary commit" is ready for merge.
